node/src/components/storage.rs

```diff
@@ -989,3 +1029,52 @@ impl Storage {
         switch_block
     }
 }
+
+/// Utility function to check the integrity of the block_body database at bringup.
+fn check_block_body_db(env: &Environment, block_body_db: &Database) -> Result<(), LmdbExtError> {
+    info!("Checking block body db");
+    let body_txn = env.begin_ro_txn()?;
+    let mut cursor = body_txn.open_ro_cursor(*block_body_db)?;
+
+    for (raw_key, raw_val) in cursor.iter() {
+        let body: BlockBody = lmdb_ext::deserialize(raw_val)?;
+        assert_eq!(
+            raw_key,
+            body.hash().as_ref(),
+            "found corrupt block body in database"
+        );
+    }
+    info!("block body db check complete");
+    drop(cursor);
+    drop(body_txn);
```

Comment:
Do we have to call `drop` explicitly? Shouldn't Rust take care of it and call `Drop::drop` on the `cursor` and `body_txn` as we exit the method?

Reply:
Exiting method w/o explicit `drop`